src/uct/base/uct_iface.c

```
@@ -332,35 +332,35 @@ ucs_status_t uct_set_ep_failed(ucs_class_t *cls, uct_ep_h tl_ep,
-    ops->ep_atomic64_fetch  = (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_atomic32_fetch  = (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_tag_eager_short = (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_tag_eager_bcopy = (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_tag_eager_zcopy = (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_tag_rndv_zcopy  = (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_tag_rndv_cancel = (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_tag_rndv_request= (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_pending_add     = (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_pending_purge   = uct_ep_failed_purge;
-    ops->ep_flush           = (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_fence           = (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_check           = (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_connect_to_ep   = (void*)ucs_empty_function_return_ep_timeout;
-    ops->ep_destroy         = uct_ep_failed_destroy;
-    ops->ep_get_address     = (void*)ucs_empty_function_return_ep_timeout;
+    ops->ep_put_short        = (uct_ep_put_short_func_t)ucs_empty_function_return_ep_timeout;
+    ops->ep_put_bcopy        = (uct_ep_put_bcopy_func_t)ucs_empty_function_return_bc_ep_timeout;
+    ops->ep_put_zcopy        = (uct_ep_put_zcopy_func_t)ucs_empty_function_return_ep_timeout;
+    ops->ep_get_short        = (uct_ep_get_short_func_t)ucs_empty_function_return_ep_timeout;
+    ops->ep_get_bcopy        = (uct_ep_get_bcopy_func_t)ucs_empty_function_return_ep_timeout;
+    ops->ep_get_zcopy        = (uct_ep_get_zcopy_func_t)ucs_empty_function_return_ep_timeout;
+    ops->ep_am_short         = (uct_ep_am_short_func_t)ucs_empty_function_return_ep_timeout;
+    ops->ep_am_bcopy         = (uct_ep_am_bcopy_func_t)ucs_empty_function_return_bc_ep_timeout;
```

does it means we also need to change static initializers in all transports?

---

strict casting is required in case if function type doesn't correspond to pointer declaration. there is way to suppress this error/warning in pgcc18, but it could lead missing errors

---

so the answer is yes?

---

yes, most of static initializers have to be updated

---

ok, sounds fine to me, could also be 1 step closer to -Wpedantic